Added release notes up to 10.15.246.0 (#77)
SoftwareAG/c8y-releasenotes: content/release-10-16-0/platform-10-16-0-bundle/10-16-0-0.md

<td>
Inventory roles</td>
<td>For users with inventory role access only, the performance for GET requests for operations has been improved. The feature is hidden behind a feature toggle which is already used for other improvements. The tenant option to enable/disable the toggle has the following payload: <code>"category": "configuration", "key": "acl.algorithm-version", "value": "OPTIMIZED"/"LEGACY"</code>. For details, see the <a href="https://www.cumulocity.com/api" class="no-ajaxy">Cumulocity IoT OpenAPI Specification</a>.</td>
<td>
MTM-42354</td>
</tr>

<tr>
<td>Inventory roles</td>

> <td>Inventory roles algorithms with improved performance are now enabled on all tenants by default.

Author: ```suggestion
<td>The improved performance option for inventory roles is now enabled on all tenants by default.

```

> In previous versions they were disabled, and customer needed to turn on feature per-tenant using tenant option or Operation team needed to turn on feature per environment.</td>

Reviewer: ```suggestion
In previous versions they were disabled, and customers needed to enable the feature for each tenant using a tenant option. Alternatively, the operations team could enable the feature per environment.</td>
```
maybe even remove the last sentence alltogether. it's more details than necessary for the side note here.

Author: There is definitely something wrong with this entry, need to double-check as this is not reviewed at all :-)

<tr>
<td>
REST API</td>
<td>The <code>detail()</code> and <code>delete()</code> method of the @c8y/client MeasurementService has been marked as deprecated as it is not supported with the new timeseries.</td>
<td>
MTM-48529</td>

<tr>
<td>REST API</td>
<td>The `activeVersionId` and the hosted application's manifest will now be updated when updating the `latest` version of an application.</td>
<td>MTM-48430</td>
</tr>

<tr>
<td>REST API</td>
<td>The REST method <code>GET /measurement/measurements/{id}</code> is no longer supported for timeseries measurements.</td>
<td>MTM-48344</td>
</tr>

<tr>
<td>Security</td>

> <td>An attribute was added to all external application links, that prevents tab nabbing (a security vulnerability that could be used to gain the user's browser session).</td>

Reviewer: We could maybe remove the parentheses here. tab nagging is probably not commonly known.

Author: Will do

> <td>An attribute was added to all external application links, that prevents tab nabbing (a security vulnerability that could be used to gain the user's browser session).</td>

Author: ```suggestion
<td>An attribute was added to all external application links, that prevents tab nabbing, a security vulnerability that could be used to gain the user's browser session.</td>

```

<td>MTM-48156</td>
</tr>

<tr>
<td>Single Sign On</td>
<td>The single sign-on (SSO) configuration page has been extended by a new access mapping option. Instead of a checkbox which sets the mapping during each login or only during user creation, a radio button has been added. Now three options are available:
<br>- Perform access mapping for every login for all privileges (admin can not edit SSO user privileges)
<br>- Perform access mapping only on user creation (admin can edit SSO user privileges)  

> <br>- Perform access mapping for every login for those privileges listed in the configuration (admin can edit SSO user privileges)</td>

Reviewer: there's no periods at the end of the list items. Intended?

Author: I think for release notes purposes this is ok, but I could add them?

MTM-48010</td>
</tr>

<tr>
<td>Core platform</td>

> <td>The managed object delete will no longer be asynchronous. The request will be blocked until the managed object and the related data is deleted. When the status code 2xx is returned by the platform then this means that all requested data has been deleted. Prior to this change, the platform returned the status code 2xx but the operation was triggered as asynchronous in the background. When the operation was unsuccessful the managed object was left.

Reviewer: ```suggestion
<td>Deleting managed objects is no longer asynchronous. Delete requests will be blocked until the managed objects and the related data are deleted. Once status code 2xx is returned by the platform, all related data have been deleted. Prior to this change, the platform returned status code 2xx immediately and the operation was running asynchronously. When the operation was unsuccessful the managed objects remained.
```

<td>
MTM-47097</td>

<tr>
<td>REST API</td>

> <td>Fixed the behaviour of the <code>withParents=false</code> parameter for the REST managed object resource, that is <code>GET /inventory/managedObjects?withParents=false</code>.

Reviewer: ```suggestion
<td>Fixed the behavior of the <code>withParents=false</code> parameter for the REST managed object resource, that is <code>GET /inventory/managedObjects?withParents=false</code>.
```

> When the false parameter value is provided, no parents are returned.</td>

Reviewer: ```suggestion
If the parameter is set to `false`, no parents are returned.</td>
```
